tests/integration/src/test/scala/mesosphere/marathon/core/election/CuratorElectionStreamTest.scala

```diff
@@ -160,7 +160,35 @@ class CuratorElectionStreamTest extends AkkaUnitTest with Inside with ZookeeperS
 
     inside(failure) {
       case ex: IllegalStateException =>
-        ex.getMessage shouldBe "Multiple election participants have the same ID: duplicate-host. This is not allowed."
+        ex.getMessage shouldBe "Multiple election participants have the same id: duplicate-host. This is not allowed."
     }
   }
+
+  "CuratorElectionStream quickly emits uncertainty about current leader during connection troubles" in withFixture { f =>
+    val (cancellable, leader) = CuratorElectionStream(
+      f.metrics, f.client, f.leaderPath, 5000.millis, "host:8080", f.electionEC)
+      .toMat(Sink.queue())(Keep.both)
+      .run
+    Given("an elected leader")
+    leader.pull().futureValue shouldBe Some(LeadershipState.ElectedAsLeader)
+
+    When("we stop the Zookeeper server")
+    zkServer.stop()
```

Comment:
This can only be the last test in the suite now.

Reply:
- [ ] lol, good point. We'll put the start in a finally block.